### ISSUE TITLE
dependent methods are run in parallel

### ISSUE THREAD

**Issue by maco-the-elder** [cbeust/testng#1411]
### TestNG Version
6.11

### Expected behavior
Dependendent test methods (annotated with dependsOnMethods) are run sequentially. Methods that depend on failed methods are skipped.

### Actual behavior
Under certain circumstances (see below) dependent methods are run in parallel. 

### Is the issue reproductible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [x] IntelliJ
- [ ] NetBeans

### Test case sample

I condesed the problem using this minimal example. Class DependsOnTests contains 3 interdependent methods (A <- B <- C) and an independent method (X). B always fails, so C shouldn't run at all.

```java
package playground;

import org.testng.Assert;
import org.testng.annotations.Test;

import java.util.logging.Logger;

public class DependsOnTests {

    private final Logger logger = Logger.getLogger(getClass().getName());

    @Test
    public void testA() throws InterruptedException {
        logger.info("running testA on Thread " + Thread.currentThread().getName());
        Thread.sleep(1000);
    }

    @Test(dependsOnMethods = "testA")
    public void testB() throws InterruptedException {
        logger.info("running testB on Thread " + Thread.currentThread().getName());
        Thread.sleep(1000);
        Assert.fail("failing on purpose");
    }

    @Test(dependsOnMethods = "testB")
    public void testC() throws InterruptedException {
        logger.info("running testC on Thread " + Thread.currentThread().getName());
        Thread.sleep(1000);
        logger.warning("this point should never be reached, since method should be skipped");
    }

    @Test
    public void testX() throws InterruptedException {
        logger.info("running testX on Thread " + Thread.currentThread().getName());
        Thread.sleep(1000);
    }

}
```
The Tests are run using following suite.
```xml
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >

<suite name="Playground Tests">

    <test name="TestNG Glitch" parallel="methods">
        <classes>
            <class name="playground.DependsOnTests"/>
        </classes>
    </test>

</suite>
```
which gives this output:
```
Running TestSuite
Apr 07, 2017 4:26:58 PM playground.DependsOnTests testX
INFORMATION: running testX on Thread pool-1-thread-2
Apr 07, 2017 4:26:58 PM playground.DependsOnTests testA
INFORMATION: running testA on Thread pool-1-thread-1
Apr 07, 2017 4:26:59 PM playground.DependsOnTests testB
INFORMATION: running testB on Thread pool-1-thread-3
Apr 07, 2017 4:26:59 PM playground.DependsOnTests testC
INFORMATION: running testC on Thread pool-1-thread-4
Apr 07, 2017 4:27:00 PM playground.DependsOnTests testC
WARNUNG: this point should never be reached, since method should be skipped
Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.521 sec <<< FAILURE!

Results :

Failed tests:   testB(playground.DependsOnTests): failing on purpose

Tests run: 4, Failures: 1, Errors: 0, Skipped: 0
```
As you can see, all methods are run in parallel. As a result, C does not get skipped as expected.
Please note that the depencency graph works as expected, if
- there are no additional test methods (like X in the example)
- parallel != "methods"|"true"
- Test-NG version is 6.9.10 or lower

**Comment by juherr** [comment]
Thanks for the report. 

We already fixed some parallel issues: https://github.com/cbeust/testng/blob/master/CHANGES.txt
Could you try again with `6.11.1-SNAPSHOT` and tell us if the issue still exist?

**Comment by maco-the-elder** [comment]
As suggested, I have tried 6.11.1-SNAPSHOT (1e1d341f46a69f679a12eedaf742ad22cafd3d1e). I can verify the the bug doesn't occur anymore.
I appreciate your brilliant work :) Thanks.

**Comment by juherr** [comment]
Fixed in next release :)

**Comment by adrian-herscu** [comment]
happens with 6.14.3

**Comment by krmahadevan** [comment]
Can you please help share a simple standalone test that can be used to
reproduce the issue and perhaps in a new issue ?

**Comment by MicahLC** [comment]
Hey @adrian-herscu , I think @krmahadevan means you should create a test in the TestNG project that reproduces the issue (and, presumably, fails).

**Comment by juherr** [comment]
@MicahLC Yes, it's always awesome. But samples are already a good start.

**Comment by adrian-herscu** [comment]
The sample is in the ticket referenced above.
Been answered that the group dependencies feature does not work in parallel mode.
